pkg/build/controller/image_change_controller.go

```diff
@@ -78,12 +76,18 @@ func (c *ImageChangeController) HandleImageRepo() {
 			glog.V(4).Infof("Running build for buildConfig %s in namespace %s", config.Name, config.Namespace)
 			b := buildutil.GenerateBuildFromConfig(config, nil, imageSubstitutions)
 			if err := c.BuildCreator.Create(config.Namespace, b); err != nil {
-				glog.V(2).Infof("Error starting build for buildConfig %v: %v", config.Name, err)
+				return fmt.Errorf("Error starting build for buildConfig %v: %v, will be retried", config.Name, err)
 			} else {
 				if err := c.BuildConfigUpdater.Update(config); err != nil {
+					// This is not a retryable error because the build has been created.  The worst case
+					// outcome of not updating the buildconfig is that we might rerun a build for the
+					// same "new" imageid change in the future, which is better than guaranteeing we
+					// run the build 2+ times by retrying it here.
 					glog.V(2).Infof("Error updating buildConfig %v: %v", config.Name, err)
+					return ImageChangeControllerFatalError(fmt.Sprintf("Error updating buildConfig %v with new LastTriggeredImageID: %v", config.Name, err))
```

Comment:
This class should be a struct wrapping an error (so that you preserve the original error) and the string (for the reason).  Doing the sprintf here loses info.